File: Softuni HomeWork/C# Advanced/06. Linq/HomeWork/7. Functional-Programming-Homework/FunctionalProgramming/Student/Student.cs
namespace Student
{
    using System.Collections.Generic;

    public class Student
    {
        public Student(string firstName, string lastName, int age, int facultyNumber, string phone, string email, int groupNumber, IList<int> marks)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
            this.Age = age;
            this.FacultyNumber = facultyNumber;
            this.Phone = phone;
            this.Email = email;
            this.GroupNumber = groupNumber;
            this.Marks = marks;
        }

        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Age { get; set; }
        public int FacultyNumber { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public int GroupNumber { get; set; }
        public IList<int> Marks { get; set; }
    }
}

File: Express.js/04.MongoDbMongoose/mongo-db-demo/models/Student.js
const mongoose = require('mongoose')
const ObjectId = mongoose.Schema.ObjectId
const studentSchema = new mongoose.Schema({
  firstName: { type: String, required: true },
  lastName: { type: String, required: true },
  facultyNumber: { type: String, required: true, unique: true },
  age: { type: Number },
  courses: [{type: ObjectId, ref: 'Course'}],
  grades: { type: Array }
})
studentSchema.methods.getInfo = function () {
  return `I am ${this.firstName} ${this.lastName}`
} // Additional methods
studentSchema.virtual('fullName').get(function () {
  return this.firstName + ' ' + this.lastName
}) // virtual props

studentSchema.path('firstName').validate(function () {
  return this.firstName.length >= 2 && this.firstName.length <= 10
}, 'First name must be between 2 and 10 symbols long!')

module.exports = mongoose.model('Student', studentSchema)
